### PR TITLE
[Accuracy diff No.113] Fix accuracy diff for paddle.nn.functional.conv2d API

### PR DESCRIPTION
**Pull request by co63oc**
paddle.nn.functional.conv2d
使用 --test_amp=True 选项PaddleAPITest测试通过

image

### CODE REVIEW

**Comment by cangtianhuang** [comment]
按理说此配置不属于 amp 配置，所有 Tensor 的类型都是 float32：
```bash
paddle.nn.functional.conv2d(Tensor([8, 128, 256, 256],"float32"), Tensor([128, 128, 3, 3],"float32"), bias=None, stride=1, padding=1, )
```
paddlepaddle-gpu 3.0.0.dev20250702 测试未通过，但感觉误差可以接受：
image

**Comment by co63oc** [comment]
> 按理说此配置不属于 amp 配置，所有 Tensor 的类型都是 float32：
> 
> ```shell
> paddle.nn.functional.conv2d(Tensor([8, 128, 256, 256],"float32"), Tensor([128, 128, 3, 3],"float32"), bias=None, stride=1, padding=1, )
> ```
> 
> paddlepaddle-gpu 3.0.0.dev20250702 测试未通过，但感觉误差可以接受： image

那这应该怎么修改，应该不是代码问题

**Comment by co63oc** [comment]
> > > 按理说此配置不属于 amp 配置，所有 Tensor 的类型都是 float32：
> > > ```shell
> > > paddle.nn.functional.conv2d(Tensor([8, 128, 256, 256],"float32"), Tensor([128, 128, 3, 3],"float32"), bias=None, stride=1, padding=1, )
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > paddlepaddle-gpu 3.0.0.dev20250702 测试未通过，但感觉误差可以接受： image
> > 
> > 
> > 那这应该怎么修改，应该不是代码问题
> 
> 反复测了一下，精度误差的位置均在 (*, *, 0, 0) 处，： image
> 
> 而且该配置的参数与 torch 是严格对齐的，感觉可以先保留此精度问题🤔

好的，后面有时间再研究，或者哪位同学感兴趣继续查看
